crates/libm-test/benches/random.rs

```diff
@@ -0,0 +1,122 @@
+use criterion::{criterion_main, Criterion};
+use libm_test::gen::random;
+use libm_test::TupleCall;
+use std::hint::black_box;
+use std::time::Duration;
+
+/// Benchmark with this many items to get a variety
+const BENCH_ITER_ITEMS: usize = if cfg!(feature = "short-benchmarks") {
+    50
+} else {
+    500
+};
+
+macro_rules! musl_rand_benches {
+    (
+        fn_name: $fn_name:ident,
+        CFn: $CFn:ty,
+        CArgs: $CArgs:ty,
+        CRet: $CRet:ty,
+        RustFn: $RustFn:ty,
+        RustArgs: $RustArgs:ty,
+        RustRet: $RustRet:ty,
+        fn_extra: $skip_on_i586:expr,
+    ) => {
+        paste::paste! {
+            fn [< musl_bench_ $fn_name >](c: &mut Criterion) {
+                let fname = stringify!($fn_name);
+                let benchvec: Vec<_> = random::get_test_cases::<$RustArgs>(fname)
+                    .take(BENCH_ITER_ITEMS)
+                    .collect();
+
+                // Perform a sanity check that we are benchmarking the same thing
+                // Don't test against musl if it is not available
+                #[cfg(feature = "build-musl")]
+                for input in benchvec.iter().copied() {
+                    use anyhow::Context;
+                    use libm_test::{CheckBasis, CheckCtx, CheckOutput};
+
+                    const QUICKCHECK_ULP: u32 = 10;
+
+                    if cfg!(x86_no_sse) && $skip_on_i586 {
+                        break;
+                    }
+
+                    let ctx = CheckCtx {
+                        ulp: QUICKCHECK_ULP,
+                        fname,
+                        basis: CheckBasis::Musl
+                    };
```

Comment:
```suggestion
                    let ctx = CheckCtx::new(
                        QUICKCHECK_ULP,
                        fname,
                        CheckBasis::Musl
                    );
```
Needed for this to compile with feature `build-musl`

Reply:
Oops, I forgot I need to do another iteration of updates to everything against the test-refactoring branch. Thanks

Reply:
This should be fixed now.

Feel free to take a look at the other PRs in this series (https://github.com/rust-lang/libm/pull/300, https://github.com/rust-lang/libm/pull/311), I think what I have is _pretty_ reasonable but I wouldn't mind some extra reviews.